### ISSUE TITLE
Latest version (0.14.3) fails to build libxslt module on linux

### ISSUE THREAD

**Issue by tornillo** [gagern/libxmljs#2]
logs

``` bash
[t@dev test]$ npm i libxslt
/
> libxmljs-mt@0.14.3 install /home/t/test/node_modules/libxslt/node_modules/libxmljs-mt
> node-gyp rebuild

make: Entering directory `/home/t/test/node_modules/libxslt/node_modules/libxmljs-mt/build'
  CC(target) Release/obj.target/libxml/vendor/libxml/buf.o
  CC(target) Release/obj.target/libxml/vendor/libxml/catalog.o
  CC(target) Release/obj.target/libxml/vendor/libxml/chvalid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/dict.o
  CC(target) Release/obj.target/libxml/vendor/libxml/encoding.o
../vendor/libxml/encoding.c: In function ‘xmlCharEncCloseFunc’:
../vendor/libxml/encoding.c:2856:12: warning: variable ‘handler_in_list’ set but not used [-Wunused-but-set-variable]
     int i, handler_in_list = 0;
            ^
  CC(target) Release/obj.target/libxml/vendor/libxml/entities.o
  CC(target) Release/obj.target/libxml/vendor/libxml/error.o
  CC(target) Release/obj.target/libxml/vendor/libxml/globals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/hash.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLparser.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLtree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/legacy.o
  CC(target) Release/obj.target/libxml/vendor/libxml/list.o
  CC(target) Release/obj.target/libxml/vendor/libxml/parser.o
  CC(target) Release/obj.target/libxml/vendor/libxml/parserInternals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/pattern.o
  CC(target) Release/obj.target/libxml/vendor/libxml/relaxng.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX2.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX.o
  CC(target) Release/obj.target/libxml/vendor/libxml/tree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/threads.o
  CC(target) Release/obj.target/libxml/vendor/libxml/uri.o
  CC(target) Release/obj.target/libxml/vendor/libxml/valid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xinclude.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xlink.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlIO.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmemory.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmodule.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlreader.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlregexp.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlsave.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemas.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemastypes.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlstring.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlunicode.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlwriter.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
../vendor/libxml/xpath.c: In function ‘xmlXPathNodeCollectAndTest’:
../vendor/libxml/xpath.c:12406:28: warning: comparison between ‘xmlElementType’ and ‘xmlXPathTypeVal’ [-Wenum-compare]
       } else if (cur->type == type) {
                            ^
In file included from ../vendor/libxml/xpath.c:476:0:
../vendor/libxml/xpath.c: At top level:
../vendor/libxml/trionan.c:218:1: warning: ‘trio_is_negative’ defined but not used [-Wunused-function]
 trio_is_negative
 ^
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
../vendor/libxml/xpointer.c: In function ‘xmlXPtrNewRangeNodePoint’:
../vendor/libxml/xpointer.c:451:21: warning: comparison between ‘xmlElementType’ and ‘enum <anonymous>’ [-Wenum-compare]
     if (start->type != XPATH_POINT)
                     ^
  AR(target) Release/obj.target/vendor/xml.a
  COPY Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
  CXX(target) Release/obj.target/xmljs/src/xml_attribute.o
  CXX(target) Release/obj.target/xmljs/src/xml_document.o
  CXX(target) Release/obj.target/xmljs/src/xml_element.o
  CXX(target) Release/obj.target/xmljs/src/xml_comment.o
  CXX(target) Release/obj.target/xmljs/src/xml_namespace.o
  CXX(target) Release/obj.target/xmljs/src/xml_node.o
  CXX(target) Release/obj.target/xmljs/src/xml_sax_parser.o
  CXX(target) Release/obj.target/xmljs/src/xml_syntax_error.o
  CXX(target) Release/obj.target/xmljs/src/xml_xpath_context.o
  SOLINK(target) Release/obj.target/xmljs.node
  COPY Release/xmljs.node
make: Leaving directory `/home/t/test/node_modules/libxslt/node_modules/libxmljs-mt/build'

> libxslt@0.5.2 install /home/t/test/node_modules/libxslt
> node-gyp rebuild

make: Entering directory `/home/t/test/node_modules/libxslt/build'
  CC(target) Release/obj.target/libexslt/deps/libxslt/libexslt/common.o
In file included from ../deps/libxslt/libexslt/common.c:2:0:
../deps/libxslt/libexslt/libexslt.h:19:31: fatal error: libxml/xmlversion.h: No such file or directory
 #include <libxml/xmlversion.h>
                               ^
compilation terminated.
make: *** [Release/obj.target/libexslt/deps/libxslt/libexslt/common.o] Error 1
make: Leaving directory `/home/t/test/node_modules/libxslt/build'

```

**Comment by gagern** [comment]
`make -C build V=1` reports (with line breaks added by me):

```
cc '-DHAVE_CONFIG_H' '-DLIBXSLT_STATIC' '-DWITH_MODULES' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DNDEBUG'
 -I../deps/libxslt
 -I../deps/libxslt.config/linux/x64
 -I/node-libxslt/node_modules/libxmljs-mt/src
 -I/node-libxslt/node_modules/libxmljs-mt/vendor/libxml/include
 -I/usr/include/nodejs/src
 -I/usr/include/nodejs/deps/uv/include
 -I/usr/include/nodejs/deps/v8/include
 -fPIC -Wall -Wextra -Wno-unused-parameter -pthread -m64 -O2 -fno-strict-aliasing -fno-tree-vrp -fno-omit-frame-pointer  -MMD -MF ./Release/.deps/Release/obj.target/libexslt/deps/libxslt/libexslt/common.o.d.raw  -c -o Release/obj.target/libexslt/deps/libxslt/libexslt/common.o ../deps/libxslt/libexslt/common.c
In file included from ../deps/libxslt/libexslt/common.c:2:0:
../deps/libxslt/libexslt/libexslt.h:19:31: fatal error: libxml/xmlversion.h: No such file or directory
 #include <libxml/xmlversion.h>
                               ^
```

So we have `libxmljs-mt/vendor/libxml/include` in there, but not `libxmljs-mt/vendor/libxml.conf/include` as I would have expected from [the gyp file](https://github.com/gagern/libxmljs/blob/v0.14.3-mt/vendor/libxml.gyp#L53). But perhaps that's because that setting describes a _direct_ dependent project, in this case the libxmljs top level.

So where does the line above come from? It comes [from libxslt itself](https://github.com/albanm/node-libxslt/blob/5e75c7ffd371a6ea3dec4118fc6e1e903a6a144d/common.gypi#L7). It seems that my version bump from 0.14.2 to 0.14.3 introduced a backwards-incompatible change due to a changed file path. I don't know whether there is anything I can do inside libxmljs to declare my include paths to node-libxslt. But the thing can certainly be addressed by modifying node-libxslt itself, as discussed in https://github.com/albanm/node-libxslt/issues/20.

I might also try convincing npm to include the file in its old location in the published version. Not sure how to achieve that, will require some research.

**Comment by gagern** [comment]
Unfortunately, 0fe2b629919a6b7fd9d7bf35826f191e2279be42 didn't work as I'd have hoped. I wish I could try out packaging without actually uploading tons of versions to the npmjs registry.

**Comment by gagern** [comment]
Fixed by d5455d4f5cedce136994ff9c0fda20cfcde758f3 released in libxmljs-mt@0.14.5.

**Comment by tornillo** [comment]
Great! Thank you.
